### PR TITLE
Adds stablehlo_batch_norm_grad op

### PR DESCRIPTION
**Pull request by amrinfathima-mcw**
- Adds stablehlo batch_norm_grad op implementation 
- Adds stablehlo batch_norm_grad op unit tests

### CODE REVIEW

File: tensorflow/lite/kernels/stablehlo_batch_norm_grad.cc [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+      context, TfLiteIntArrayEqual(operand->dims, grad_operand->dims), true);
+
+  TF_LITE_ENSURE_EQ(context, TfLiteIntArrayEqual(scale->dims, mean->dims),
+                    true);
+  TF_LITE_ENSURE_EQ(context, TfLiteIntArrayEqual(scale->dims, variance->dims),
+                    true);
+  TF_LITE_ENSURE_EQ(context, TfLiteIntArrayEqual(scale->dims, grad_scale->dims),
+                    true);
+  TF_LITE_ENSURE_EQ(context,
+                    TfLiteIntArrayEqual(scale->dims, grad_offset->dims), true);
```
Comment (amrin1908) [inline tensorflow/lite/kernels/stablehlo_batch_norm_grad.cc:364]: has been decided to not go with this approach

File: tensorflow/lite/kernels/stablehlo_batch_norm_grad.cc [hunk 2]
```diff
@@ -204,6 +212,66 @@ TfLiteStatus PrepareTemporaries(TfLiteContext* context, TfLiteNode* node,
...
+    TfLiteIntArray* i5_shape = TfLiteIntArrayCreate(operand->dims->size);
+    for (int i = 0; i < operand->dims->size; ++i) {
+      i5_shape->data[i] = operand->dims->data[i];
+    }
+    node->temporaries->data[13] = data->scratch_tensor_index + 13;
+    TfLiteTensor* i5;
+    TF_LITE_ENSURE_OK(context, GetTemporarySafe(context, node, 13, &i5));
+    i5->type = kTfLiteInt8;
+    i5->allocation_type = kTfLiteArenaRw;
+    TF_LITE_ENSURE_OK(context, context->ResizeTensor(context, i5, i5_shape));
```
Comment (amrin1908) [inline tensorflow/lite/kernels/stablehlo_batch_norm_grad.cc:262]: check int16

Comment (amrin1908) [reply tensorflow/lite/kernels/stablehlo_batch_norm_grad.cc:262]: okay

Comment (amrinfathima-mcw) [reply tensorflow/lite/kernels/stablehlo_batch_norm_grad.cc:262]: okay

**Pull request closed without merging**